dashboard/src/actions/authActions.js

```diff
@@ -2,114 +2,58 @@ import * as APP_ROUTES from "utils/routeConstants";
+      reject(new Error('Something went wrong'));
+    } else {
+      setTimeout(check, 250, resolve, reject);
     }
-};
-
-export const makeLoginRequest =
-  (details, navigate) => async (dispatch, getState) => {
-    try {
-      dispatch({ type: TYPES.LOADING });
-      // empty the alerts
-      dispatch({
-        type: TYPES.USER_NOTION_ALERTS,
-        payload: [],
-      });
-      const endpoints = getState().apiEndpoint.endpoints;
-      const response = await API.post(endpoints?.api?.login, {
-        ...details,
-      });
-      if (response.status === 200 && Object.keys(response.data).length > 0) {
-        const keepUser = getState().userAuth.keepLoggedIn;
-        const expiryTime = keepUser
-          ? CONSTANTS.EXPIRY_KEEPUSER_DAYS
-          : CONSTANTS.EXPIRY_DEFAULT_DAYS;
-        Cookies.set("isLoggedIn", true, { expires: expiryTime });
-        Cookies.set("token", response.data?.auth_token, {
-          expires: expiryTime,
-        });
-        Cookies.set("username", response.data?.username, {
-          expires: expiryTime,
-        });
-        const loginDetails = {
-          isLoggedIn: true,
-          token: response.data?.auth_token,
-          username: response.data?.username,
-        };
-        await dispatch({
-          type: TYPES.SET_LOGIN_DETAILS,
-          payload: loginDetails,
-        });
-
-        navigate(APP_ROUTES.OVERVIEW);
+  };
+  return new Promise((resolve, reject) => {
+    check(resolve, reject);
+  });
+}
 
-        dispatch(showToast(SUCCESS, "Logged in successfully!"));
-      }
-      dispatch({ type: TYPES.COMPLETED });
-    } catch (error) {
-      const alerts = getState().userAuth.alerts;
-      let alert = {};
-      if (error?.response) {
-        alert = {
-          title: error?.response?.data?.message,
-          key: uid(),
-        };
-        dispatch(toggleLoginBtn(true));
-      } else {
-        alert = {
-          title: error?.message,
-          key: uid(),
-        };
-        dispatch({ type: TYPES.NETWORK_ERROR });
-      }
-      alerts.push(alert);
+// Perform some house keeping when the user logs in
+export const authCookies = () => async (dispatch, getState) => {
+    await waitForKeycloak(getState);
+    const keycloak = getState().apiEndpoint.keycloak;
+    if (keycloak.authenticated){
+      Cookies.set("isLoggedIn", true, { expires: keycloak.refreshTokenParsed?.exp });
+      Cookies.set("username", keycloak.tokenParsed?.preferred_username, { expires: keycloak.refreshTokenParsed?.exp });
+      const userPayload = {
+        "username": keycloak.tokenParsed?.preferred_username,
+        "email": keycloak.tokenParsed?.email,
+        "first_name": keycloak.tokenParsed?.given_name,
+        "last_name": keycloak.tokenParsed?.family_name,
+      };
       dispatch({
-        type: TYPES.USER_NOTION_ALERTS,
-        payload: alerts,
+        type: TYPES.GET_USER_DETAILS,
+        payload: userPayload,
       });
```

Comment:
`GET_USER_DETAILS` is misnamed action type (you're using it to _**set**_ the user details!); please consider renaming it.

You should check on the rest of the auth reducer action types and tear out any reducer cases which are no longer used (along with their infrastructure...like `userDetails_copy`?)...at which point `SET_USER_DETAILS` might become available for use here....